lms/djangoapps/badges/models.py

```diff
@@ -0,0 +1,280 @@
+"""
+Database models for the badges app
+"""
+import re
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.importlib import import_module
+from django.utils.translation import ugettext_lazy as _
+from lazy import lazy
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from config_models.models import ConfigurationModel
+from xmodule_django.models import CourseKeyField
+from jsonfield import JSONField
+
+
+def validate_badge_image(image):
+    """
+    Validates that a particular image is small enough, of the right type, and square to be a badge.
+    """
+    if image.width != image.height:
+        raise ValidationError(_(u"The badge image must be square."))
+    if not image.size < (250 * 1024):
+        raise ValidationError(_(u"The badge image file size must be less than 250KB."))
+
+
+class BadgeClass(models.Model):
+    """
+    Specifies a badge class to be registered with a backend.
+    """
+    slug = models.SlugField(max_length=255)
+    issuing_component = models.SlugField(max_length=50, default='', blank=True)
```

Comment:
Nit: The empty string is the default anyway.  (I'm in general all for being expicit, but in this particular case it really doesn't seem to add anything to the readability of the code.)